OutOfSchool/OutOfSchool.WebApi.Tests/Services/NotificationServiceTests.cs

```diff
@@ -0,0 +1,270 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using NUnit.Framework;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.Tests.Common;
+using OutOfSchool.WebApi.Config;
+using OutOfSchool.WebApi.Hubs;
+using OutOfSchool.WebApi.Models.Notifications;
+using OutOfSchool.WebApi.Services;
+using OutOfSchool.WebApi.Util;
+using OutOfSchool.WebApi.Util.Mapping;
+
+namespace OutOfSchool.WebApi.Tests.Services;
+
+[TestFixture]
+public class NotificationServiceTests
+{
+    private INotificationService notificationService;
+    private Mock<INotificationRepository> notificationRepositoryMock;
+    private Mock<ILogger<NotificationService>> loggerMock;
+    private Mock<IStringLocalizer<SharedResource>> localizerMock;
+    private IMapper mapper;
+    private Mock<IHubContext<NotificationHub>> notificationHubMock;
+    private Mock<IOptions<NotificationsConfig>> notificationsConfigMock;
+
+    private string userId;
+    private List<Notification> notifications;
+
+    [SetUp]
+    public void SetUp()
+    {
+        notificationRepositoryMock = new Mock<INotificationRepository>();
+        loggerMock = new Mock<ILogger<NotificationService>>();
+        localizerMock = new Mock<IStringLocalizer<SharedResource>>();
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        notificationHubMock = new Mock<IHubContext<NotificationHub>>();
```

Comment:
(Redundant fields) Some dependencies used in `notificationService` - `loggerMock`, `localizerMock`, and `notificationHubMock` are not touched by tests, so could be easily dropped from fields list and even inlined in constructor. 

Having the full list of fields (as it is now) adds extra lines of code with no benefits, as set of really used mocks is more obscure. 

Ofc, one might say that this is done for the future tests writing, but it's too easy to extract a variable/field using VS refactoring, so it could really be done when needed, and there is no necessity for premature "optimisation".